### DISCUSSION TITLE
Extending config from `~/.ssh/config`

### DISCUSSION THREAD

**Discussion by seahindeniz** [Natizyskunk/vscode-sftp#292] [Ideas]
Hi, is it possible to extend config values from the `~/.ssh/config` file where I mostly store my common details?
In that file, I can use the same configuration in my terminal and also in my VSCode's Remote extension.

Here's a commonly used config entry
```
Host test-machine
    HostName 32.23.232.32
    User ubuntu
    IdentityFile ~/.ssh/keys/key.pem
```

From the RemoteFS config, maybe we can have this format
```json5
"remotefs.remote": {
    "test-machine": {
      "scheme": "sftp",
      "extend": "ssh:test-machine", // means, from the ssh config file, use the entry named `test-machine`
      "username": "ubuntu", // in addition, we can also override the extended value too when it doesn't satisfy RemoteFS
    },
```

Each entry can should suffice to supply RemoteFS config structure

**Comment by jarrah31** [comment]
I too would like this ability please. 
I'm trying to get away from using Sublime, but they have a really good SFTP plugin that utilises my Mac's ssh config file for all connections. This means it works with options such as IdentityAgent, IdentityFile, Match exec, LocalForward, etc. 
Is there a technical reason why this SFTP plugin can't utilise the hosts local SSH capabilities?

**Comment by Natizyskunk** [comment]
@seahindeniz @jarrah31 you can refer to [this](https://github.com/Natizyskunk/vscode-sftp/wiki/Configuration#sshconfigpath). It's as simple as adding this to your sftp.json config file : 
```json
"sshConfigPath":"~/.ssh/config",
```

Let me now if this is what you was searching for.

**Comment by jarrah31** [comment]
Hi @Natizyskunk, apologies for not seeing your earlier comment and thank you for referring me to the sshConfigPath option. I saw the closure notification which helped me find your reply. 

So unfortunately I'm still having issues trying to make this work.
 
My sftp.json looks like this:

```
{
    "name": "test HASS",
    "host": "test",
    "protocol": "sftp",
    "remotePath": "/home/pi/docker/homeassistant",
    "uploadOnSave": true,
    "useTempFile": false,
    "openSsh": false,
    "sshConfigPath": "~/.ssh/config"
}
```

The following config file example works fine:
```
Host test
 Hostname 192.168.1.32
 IdentityFile ~/.ssh/test_id_rsa
 Port 22
 User pi
```
However if I use `Host *` to save me repeating lines across all my Hosts, it doesn't pick up the IdentifyFile and asks me for a password.
```
Host *
 StrictHostKeyChecking no
 IdentityFile ~/.ssh/test_id_rsa

Host test
 Hostname 192.168.1.32
 Port 22
 User pi
```
I also make heavy use of ProxyJump and Match commands, but these aren't working either. e.g
```
Match exec "/Users/me/onsubnet 192.168.1." host dsrv2
 Hostname 192.168.1.32
 Port 22
 User pi

Match exec "/Users/me/onsubnet --not 192.168.1." host dsrv2
 ProxyJump mypi
 Hostname 192.168.1.32
 User pi
 Port 22

Host mypi
 HostName my domain.com
 Port 9999
 User pi
```
I guess SFTP doesn't use the native ssh client?

As this Discussion is closed I've created a new issue so will track it there:
https://github.com/Natizyskunk/vscode-sftp/issues/433
